### ISSUE TITLE
MultiSelect clear button interactive styles

### ISSUE THREAD

**Issue by diego-codes** [carbon-design-system/carbon#2918]
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

As a mostly keyboard-only user, when I tab through the MultiSelect component, the clear selections button does not have sufficient focus (or even hover) style differences to make it clear that holds the keyboard focus.

> Is this issue related to a specific component?

Yes, the `MultiSelect`

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I think it would improve the usability of this component to provide better focus and hover styles to the clear selections button.  

> What version of the Carbon Design System are you using?

10.x

## Additional information
Tabbing through `MultiSelect`
![Kapture 2019-06-04 at 8 56 05](https://user-images.githubusercontent.com/5973294/58884777-a906d080-86a6-11e9-9ea5-52847d60fa3d.gif)

**Comment by dakahn** [comment]
Totally great catch. I'll ping @carbon-design-system/design and we can get a PR going based on their specs.

**Comment by emyarod** [comment]
related #2360 #2654

**Comment by aagonzales** [comment]
So we're looking for a hover and focus style for the tag?

**Comment by asudoh** [comment]
I think it's only about focus-style. @emyarod @diego-codes correct me if I'm wrong here.

**Comment by emyarod** [comment]
I'm using the same style for hover and focus so depending on what we go with design wise, it could be focus only or both hover and focus

**Comment by asudoh** [comment]
OK added @aagonzales as a reviewer for #2936 - @aagonzales Please feel free to delegate in case another designer should review - Thanks!
